### PR TITLE
Use dotnet runtime extension

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Closes https://github.com/tintoy/msbuild-project-tools-vscode/issues/89

I managed to test this change on a PC without .NET 6 installed. As expected, current release version of extension just refused to work, while the version after changes in this PR started without any complaints. The language server itself spawns .NET sub-processes assuming that global `dotnet` command exists and can run .NET 6 assemblies, so the extension won't fully work on PC without .NET 6. However, since this is for task reflection stuff, which is soon gonna be merged for in-process execution, I didn't do anything special for it. If we managed to get rid of sub-process spawning, we are good. Otherwise we can pass .NET path to language server via environment variables later

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     },
     "icon": "images/icon.png",
     "engines": {
-        "vscode": "^1.39.0"
+        "vscode": "^1.72.0"
```
Comment (DoctorKrolic) [inline package.json:15]: .NET runtime extension has engine version 1.72+, so I updated ours as well

File: package.json [hunk 2]
```diff
@@ -220,28 +220,29 @@
...
-        "@types/semver": "^5.3.34",
-        "@types/vscode": "^1.39.0",
+        "@types/vscode": "^1.72.0",
         "@types/which": "^1.0.28",
         "@types/xmldom": "^0.1.29",
         "chai": "^4.2.0",
         "mocha": "^6.2.2",
         "tslint": "^5.20.0",
-        "typescript": "^3.6.4"
+        "typescript": "^4.1.6"
```
Comment (DoctorKrolic) [inline package.json:238]: bump of typescript version was required for newer version of VS Code types

File: package.json [hunk 3]
```diff
@@ -220,28 +220,29 @@
...
     },
     "dependencies": {
         "mz": "^2.6.0",
         "object-path": "^0.11.4",
         "rxjs": "^5.4.2",
-        "semver": "^5.7.1",
```
Comment (DoctorKrolic) [inline package.json:242]: no longer using this library

File: src/extension/extension.ts [hunk 4]
```diff
@@ -50,21 +45,23 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
 
         await loadConfiguration();
 
-        const dotnetVersion: semver.SemVer = await dotnet.getHostVersion();
-        const canEnableLanguageService: Boolean = dotnetVersion && (dotnetVersion.compare(supportedRuntimeVersion) >= 0);
+        const dotnetExecutablePath = await dotnet.acquireRuntime(context.extension.id, progress);
```
Comment (DoctorKrolic) [inline src/extension/extension.ts:48]: `context.extension` was not available in engine version 1.39. No idea when exactly it appeared, but it is 100% there for 1.72+

**Comment by tintoy** [comment]
Nice one!

Related: tintoy/msbuild-project-tools-vscode#89

**Comment by tintoy** [comment]
Thanks!

**Pull request merged**